services/viva-ms/lambdas/personApplication.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable no-console */
+import to from 'await-to-js';
+
+import putVivaMsEvent from '../helpers/putVivaMsEvent';
+
+import vivaAdapter from '../helpers/vivaAdapterRequestClient';
+
+export async function main(event) {
+  const clientUser = event.detail.user;
```

Comment:
We should add safe check here so that contents in detail is not undefined.

```JS
function noSuchProperty(key, object) {
  // test if a property doesn't exist in the current object
  return !object.hasOwnProperty(key)
}

if(noSuchProperty('user', event.detail)) {
   // do something
 }

```

Reply:
Will be fixed in future task